test_good_dirs fails with strange error

When packaging for openSUSE (**after** applying all patches from #66, so that is not the issue) I get these errors:
```
[   28s] + PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-blue-0.8.0-0.x86_64/usr/lib/python3.9/site-packages
[   28s] + PYTHONDONTWRITEBYTECODE=1
[   28s] + pytest-3.9 --ignore=_build.python39 --ignore=_build.python310 --ignore=_build.python38 -v
[   28s] ============================= test session starts ==============================
[   28s] platform linux -- Python 3.9.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /usr/bin/python3.9
[   28s] cachedir: .pytest_cache
[   28s] rootdir: /home/abuild/rpmbuild/BUILD/blue-0.8.0, configfile: tox.ini, testpaths: blue, docs, tests
[   29s] collecting ... collected 8 items
[   29s] 
[   29s] tests/test_blue.py::test_good_dirs[config_setup] FAILED                  [ 12%]
[   29s] tests/test_blue.py::test_good_dirs[config_tox] FAILED                    [ 25%]
[   29s] tests/test_blue.py::test_good_dirs[config_blue] FAILED                   [ 37%]
[   29s] tests/test_blue.py::test_good_dirs[config_pyproject] FAILED              [ 50%]
[   29s] tests/test_blue.py::test_good_dirs[good_cases] FAILED                    [ 62%]
[   29s] tests/test_blue.py::test_bad_dirs[bad_cases] FAILED                      [ 75%]
[   29s] tests/test_blue.py::test_main PASSED                                     [ 87%]
[   29s] tests/test_blue.py::test_version PASSED                                  [100%]
[   29s] 
[   29s] =================================== FAILURES ===================================
[   29s] _________________________ test_good_dirs[config_setup] _________________________
[   29s] 
[   29s] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f702f5393a0>
[   29s] test_dir = PosixPath('/home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/config_setup')
[   29s] 
[   29s]     @pytest.mark.parametrize(
[   29s]         'test_dir',
[   29s]         [
[   29s]             'config_setup',
[   29s]             'config_tox',
[   29s]             'config_blue',
[   29s]             'config_pyproject',
[   29s]             'good_cases',
[   29s]         ],
[   29s]     )
[   29s]     def test_good_dirs(monkeypatch, test_dir):
[   29s]         test_dir = tests_dir / test_dir
[   29s]         monkeypatch.chdir(test_dir)
[   29s]         monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
[   29s]         for path in test_dir.rglob('*'):
[   29s]             path.touch()  # Invalidate file caches in Blue.
[   29s]         black.find_project_root.cache_clear()
[   29s]         with pytest.raises(SystemExit) as exc_info:
[   29s]             asyncio.set_event_loop(asyncio.new_event_loop())
[   29s]             blue.main()
[   29s] >       assert exc_info.value.code == 0
[   29s] E       assert 123 == 0
[   29s] E         +123
[   29s] E         -0
[   29s] 
[   29s] /home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/test_blue.py:32: AssertionError
[   29s] ----------------------------- Captured stderr call -----------------------------
[   29s] error: cannot format example.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] 
[   29s] Oh no! 💥 💔 💥
[   29s] 1 file would fail to reformat.
[   29s] __________________________ test_good_dirs[config_tox] __________________________
[   29s] 
[   29s] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f702f5b0070>
[   29s] test_dir = PosixPath('/home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/config_tox')
[   29s] 
[   29s]     @pytest.mark.parametrize(
[   29s]         'test_dir',
[   29s]         [
[   29s]             'config_setup',
[   29s]             'config_tox',
[   29s]             'config_blue',
[   29s]             'config_pyproject',
[   29s]             'good_cases',
[   29s]         ],
[   29s]     )
[   29s]     def test_good_dirs(monkeypatch, test_dir):
[   29s]         test_dir = tests_dir / test_dir
[   29s]         monkeypatch.chdir(test_dir)
[   29s]         monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
[   29s]         for path in test_dir.rglob('*'):
[   29s]             path.touch()  # Invalidate file caches in Blue.
[   29s]         black.find_project_root.cache_clear()
[   29s]         with pytest.raises(SystemExit) as exc_info:
[   29s]             asyncio.set_event_loop(asyncio.new_event_loop())
[   29s]             blue.main()
[   29s] >       assert exc_info.value.code == 0
[   29s] E       assert 123 == 0
[   29s] E         +123
[   29s] E         -0
[   29s] 
[   29s] /home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/test_blue.py:32: AssertionError
[   29s] ----------------------------- Captured stderr call -----------------------------
[   29s] error: cannot format example.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] 
[   29s] Oh no! 💥 💔 💥
[   29s] 1 file would fail to reformat.
[   29s] _________________________ test_good_dirs[config_blue] __________________________
[   29s] 
[   29s] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f702f5aa550>
[   29s] test_dir = PosixPath('/home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/config_blue')
[   29s] 
[   29s]     @pytest.mark.parametrize(
[   29s]         'test_dir',
[   29s]         [
[   29s]             'config_setup',
[   29s]             'config_tox',
[   29s]             'config_blue',
[   29s]             'config_pyproject',
[   29s]             'good_cases',
[   29s]         ],
[   29s]     )
[   29s]     def test_good_dirs(monkeypatch, test_dir):
[   29s]         test_dir = tests_dir / test_dir
[   29s]         monkeypatch.chdir(test_dir)
[   29s]         monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
[   29s]         for path in test_dir.rglob('*'):
[   29s]             path.touch()  # Invalidate file caches in Blue.
[   29s]         black.find_project_root.cache_clear()
[   29s]         with pytest.raises(SystemExit) as exc_info:
[   29s]             asyncio.set_event_loop(asyncio.new_event_loop())
[   29s]             blue.main()
[   29s] >       assert exc_info.value.code == 0
[   29s] E       assert 123 == 0
[   29s] E         +123
[   29s] E         -0
[   29s] 
[   29s] /home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/test_blue.py:32: AssertionError
[   29s] ----------------------------- Captured stderr call -----------------------------
[   29s] error: cannot format example.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] 
[   29s] Oh no! 💥 💔 💥
[   29s] 1 file would fail to reformat.
[   29s] _______________________ test_good_dirs[config_pyproject] _______________________
[   29s] 
[   29s] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f702f5b4b20>
[   29s] test_dir = PosixPath('/home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/config_pyproject')
[   29s] 
[   29s]     @pytest.mark.parametrize(
[   29s]         'test_dir',
[   29s]         [
[   29s]             'config_setup',
[   29s]             'config_tox',
[   29s]             'config_blue',
[   29s]             'config_pyproject',
[   29s]             'good_cases',
[   29s]         ],
[   29s]     )
[   29s]     def test_good_dirs(monkeypatch, test_dir):
[   29s]         test_dir = tests_dir / test_dir
[   29s]         monkeypatch.chdir(test_dir)
[   29s]         monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
[   29s]         for path in test_dir.rglob('*'):
[   29s]             path.touch()  # Invalidate file caches in Blue.
[   29s]         black.find_project_root.cache_clear()
[   29s]         with pytest.raises(SystemExit) as exc_info:
[   29s]             asyncio.set_event_loop(asyncio.new_event_loop())
[   29s]             blue.main()
[   29s] >       assert exc_info.value.code == 0
[   29s] E       assert 123 == 0
[   29s] E         +123
[   29s] E         -0
[   29s] 
[   29s] /home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/test_blue.py:32: AssertionError
[   29s] ----------------------------- Captured stderr call -----------------------------
[   29s] error: cannot format example.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] 
[   29s] Oh no! 💥 💔 💥
[   29s] 1 file would fail to reformat.
[   29s] __________________________ test_good_dirs[good_cases] __________________________
[   29s] 
[   29s] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f702f5eaee0>
[   29s] test_dir = PosixPath('/home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/good_cases')
[   29s] 
[   29s]     @pytest.mark.parametrize(
[   29s]         'test_dir',
[   29s]         [
[   29s]             'config_setup',
[   29s]             'config_tox',
[   29s]             'config_blue',
[   29s]             'config_pyproject',
[   29s]             'good_cases',
[   29s]         ],
[   29s]     )
[   29s]     def test_good_dirs(monkeypatch, test_dir):
[   29s]         test_dir = tests_dir / test_dir
[   29s]         monkeypatch.chdir(test_dir)
[   29s]         monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
[   29s]         for path in test_dir.rglob('*'):
[   29s]             path.touch()  # Invalidate file caches in Blue.
[   29s]         black.find_project_root.cache_clear()
[   29s]         with pytest.raises(SystemExit) as exc_info:
[   29s]             asyncio.set_event_loop(asyncio.new_event_loop())
[   29s]             blue.main()
[   29s] >       assert exc_info.value.code == 0
[   29s] E       assert 123 == 0
[   29s] E         +123
[   29s] E         -0
[   29s] 
[   29s] /home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/test_blue.py:32: AssertionError
[   29s] ----------------------------- Captured stderr call -----------------------------
[   29s] error: cannot format tdq_docstrings.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] error: cannot format decorators.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] error: cannot format demo.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] 
[   29s] Oh no! 💥 💔 💥
[   29s] 3 files would fail to reformat.
[   29s] ___________________________ test_bad_dirs[bad_cases] ___________________________
[   29s] 
[   29s] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f702f4898e0>
[   29s] test_dir = PosixPath('/home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/bad_cases')
[   29s] 
[   29s]     @pytest.mark.parametrize(
[   29s]         'test_dir',
[   29s]         ['bad_cases'],
[   29s]     )
[   29s]     def test_bad_dirs(monkeypatch, test_dir):
[   29s]         test_dir = tests_dir / test_dir
[   29s]         monkeypatch.chdir(test_dir)
[   29s]         monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
[   29s]         for path in test_dir.rglob('*'):
[   29s]             path.touch()  # Invalidate file caches in Blue.
[   29s]         black.find_project_root.cache_clear()
[   29s]         with pytest.raises(SystemExit) as exc_info:
[   29s]             asyncio.set_event_loop(asyncio.new_event_loop())
[   29s]             blue.main()
[   29s] >       assert exc_info.value.code == 1
[   29s] E       assert 123 == 1
[   29s] E         +123
[   29s] E         -1
[   29s] 
[   29s] /home/abuild/rpmbuild/BUILD/blue-0.8.0/tests/test_blue.py:49: AssertionError
[   29s] ----------------------------- Captured stderr call -----------------------------
[   29s] error: cannot format sdq_docstrings.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] error: cannot format tsq_docstrings.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] error: cannot format ssq_docstrings.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] error: cannot format demo.py: list_comments() got an unexpected keyword argument 'preview'
[   29s] 
[   29s] Oh no! 💥 💔 💥
[   29s] 4 files would fail to reformat.
[   29s] =========================== short test summary info ============================
[   29s] FAILED tests/test_blue.py::test_good_dirs[config_setup] - assert 123 == 0
[   29s] FAILED tests/test_blue.py::test_good_dirs[config_tox] - assert 123 == 0
[   29s] FAILED tests/test_blue.py::test_good_dirs[config_blue] - assert 123 == 0
[   29s] FAILED tests/test_blue.py::test_good_dirs[config_pyproject] - assert 123 == 0
[   29s] FAILED tests/test_blue.py::test_good_dirs[good_cases] - assert 123 == 0
[   29s] FAILED tests/test_blue.py::test_bad_dirs[bad_cases] - assert 123 == 1
[   29s] ========================= 6 failed, 2 passed in 0.22s ==========================
[   29s] error: Bad exit status from /var/tmp/rpm-tmp.JscPCz (%check)
```
[Complete build log](https://github.com/grantjenks/blue/files/8548103/_log.txt) with all packages used and steps taken.

What is the most interesting (in my opinion) is that `list_comments()` function (if I am not mistaken) own function of this package, so it is strange that tests call it incorrectly. Also, it is strange, that I cannot find the word `preview` in the exploded packages.

What’s going on, please?

--- a/blue/__init__.py
+++ b/blue/__init__.py
@@ -15,6 +15,14 @@ import sys
 import importlib
 import importlib.machinery
 
+
+if 'black' in sys.modules and sys.modules['black'].__file__.endswith('.so'):
+    raise RuntimeError(
+        'A mypyc-compiled version of black has already been imported. '
+        "This limits blue's ability to monkeypatch black."
+    )
+
+
 class NoMypycBlackFileFinder(importlib.machinery.FileFinder):
     def __init__(self, path: str, *loader_details) -> None:
         super().__init__(path, *loader_details)
@@ -31,16 +39,25 @@ class NoMypycBlackFileFinder(importlib.machinery.FileFinder):
 
     def find_spec(self, fullname, *args, **kw):
         if fullname == 'black' or fullname.startswith('black.'):
-            return super(NoMypycBlackFileFinder, self).find_spec(fullname, *args, **kw)
+            return super(NoMypycBlackFileFinder, self).find_spec(
+                fullname, *args, **kw
+            )
         else:
             return self.original_finder.find_spec(fullname, *args, **kw)
 
     @classmethod
     def path_hook(cls):
         return super(NoMypycBlackFileFinder, cls).path_hook(
-            (importlib.machinery.SourceFileLoader, importlib.machinery.SOURCE_SUFFIXES),
-            (importlib.machinery.SourcelessFileLoader, importlib.machinery.BYTECODE_SUFFIXES),
+            (
+                importlib.machinery.SourceFileLoader,
+                importlib.machinery.SOURCE_SUFFIXES,
+            ),
+            (
+                importlib.machinery.SourcelessFileLoader,
+                importlib.machinery.BYTECODE_SUFFIXES,
+            ),
         )
+
 
 sys.path_hooks.insert(0, NoMypycBlackFileFinder.path_hook())
 sys.path_importer_cache.clear()

--- a/tests/bad_cases/demo.py
+++ b/tests/bad_cases/demo.py
@@ -29,7 +29,7 @@ class Foo(object):
 
 
 def f(a: List[int]):
-    return 37 - a[42 - u : y ** 3]
+    return 37 - a[42 - u : y**3]
 
 
 def very_important_function(

--- a/tests/test_blue.py
+++ b/tests/test_blue.py
@@ -1,9 +1,9 @@
 import asyncio
 import pathlib
-import subprocess
+
+import blue
 
 import black
-import blue
 import pytest
 
 tests_dir = pathlib.Path(__file__).parent.absolute()
@@ -22,7 +22,7 @@ tests_dir = pathlib.Path(__file__).parent.absolute()
 def test_good_dirs(monkeypatch, test_dir):
     test_dir = tests_dir / test_dir
     monkeypatch.chdir(test_dir)
-    monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
+    monkeypatch.setattr('sys.argv', ['blue', '--check', '--diff', '.'])
     for path in test_dir.rglob('*'):
         path.touch()  # Invalidate file caches in Blue.
     black.find_project_root.cache_clear()
@@ -39,7 +39,7 @@ def test_good_dirs(monkeypatch, test_dir):
 def test_bad_dirs(monkeypatch, test_dir):
     test_dir = tests_dir / test_dir
     monkeypatch.chdir(test_dir)
-    monkeypatch.setattr('sys.argv', ['blue', '--check', '.'])
+    monkeypatch.setattr('sys.argv', ['blue', '--check', '--diff', '.'])
     for path in test_dir.rglob('*'):
         path.touch()  # Invalidate file caches in Blue.
     black.find_project_root.cache_clear()

--- a/tox.ini
+++ b/tox.ini
@@ -18,7 +18,7 @@ testpaths=blue docs tests
 commands=blue blue docs setup.py tests/test_blue.py
 
 [testenv:bluecheck]
-commands=blue --check blue docs setup.py tests/test_blue.py
+commands=blue --check --diff blue docs setup.py tests/test_blue.py
 
 [testenv:docs]
 allowlist_externals=make
@@ -56,5 +56,6 @@ ignore=
     E124
     E203
     E303
+    E402
     W503
 max-line-length=88

